pkg/ingester/ingester_v2.go

```diff
@@ -895,6 +895,14 @@ func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
 	}
 	db.DisableCompactions() // we will compact on our own schedule
 
+	// Run compaction before using this TSDB. If there is data in head that needs to be put into blocks,
+	// this will actually create the blocks. If there is no data (empty TSDB), this is a no-op.
+	// We don't do blocks compaction.
```

Comment:
It depends how you configure `-experimental.tsdb.block-ranges-period` (which is a list). If you configure multiple ranges there, then we do blocks compaction too. I wouldn't worry too much at this stage, but I would at least remove this comment because not necessarily true.